internal/http/client.go

```diff
@@ -0,0 +1,46 @@
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras/internal/credential"
+	"oras.land/oras/internal/trace"
+	"oras.land/oras/internal/version"
+)
+
+// client option struct
+type ClientOptions struct {
+	Credential      auth.Credential
```

Comment:
Just out of curiosify, could you help me explain the difference between the Credential and CredentialStore.

Reply:
1) For code here, `Credential` and `CredentialStore` will be used for [auth function](https://github.com/oras-project/oras-go/blob/main/registry/remote/auth/client.go#L71) in oras-go client, please consider the scenarios:
 - if a username and password options are provided, then we can directly construct a `Credential` and pass it in
 - if username and password are not provided, then we need to find existed auth credentials in config files, which will be taken care of `CredentialStore`.

2) This code is already refactored, please check https://github.com/oras-project/oras/blob/main/cmd/oras/internal/option/remote.go#L108-L119, which is strictly aligned to the two scenarios,